### PR TITLE
Fixing Dof exposure

### PR DESCRIPTION
**Pull request by roigcarlo**
**📝 Description**
Prevents Dof's from being deallocated while exposed to python via containers.

**🆕 Changelog**
- Fixing Dof exposure.

### CODE REVIEW

File: kratos/python/add_dofs_to_python.cpp [hunk 1]
```diff
@@ -19,89 +19,74 @@
...
-{
-namespace Python
-{
+namespace Kratos::Python {
 
 void  AddDofsToPython(pybind11::module& m)
 {
     namespace py = pybind11;
 
-    py::class_<Dof<double>>(m,"Dof")
```
Comment (roigcarlo) [inline kratos/python/add_dofs_to_python.cpp:31]: Fuck my text editor. This is the relevant change.

File: kratos/python/add_dofs_to_python.cpp [hunk 2]
```diff
@@ -19,89 +19,74 @@
...
-    .def("__lt__", [](Dof<double>& dof1, Dof<double>& dof2){return dof1<dof2; })
-    .def("__gt__", [](Dof<double>& dof1, Dof<double>& dof2){return dof1>dof2; })
-    .def("__le__", [](Dof<double>& dof1, Dof<double>& dof2){return dof1<=dof2; })
-    .def("__ge__", [](Dof<double>& dof1, Dof<double>& dof2){return dof1>=dof2; })
-    .def("__eq__", [](Dof<double>& dof1, Dof<double>& dof2){return dof1==dof2; })
-    .def("__ne__", [](Dof<double>& dof1, Dof<double>& dof2){return !(dof1==dof2); })
-    .def("__str__", PrintObject<Dof<double>>)
-    ;
-
+    py::class_<Dof<double>, std::unique_ptr<Dof<double>, py::nodelete>>(m,"Dof")
```
Comment (roigcarlo) [inline kratos/python/add_dofs_to_python.cpp:28]: This is the relevant change. Sorry for the mess... autoformat...

**Review by loumalouomega — DISMISSED** [review]

**Review by loumalouomega — DISMISSED** [review]

File: kratos/python/add_mesh_to_python.cpp [hunk 3]
```diff
@@ -530,6 +530,11 @@ void  AddMeshToPython(pybind11::module& m)
         self.EquationIdVector(ids,rProcessInfo);
         return ids;
     })
+    .def("GetDofList", [](const Element& self, const ProcessInfo& rProcessInfo){
```
Comment (philbucher) [inline kratos/python/add_mesh_to_python.cpp:533]: what about conditions?

Comment (rubenzorrilla) [reply kratos/python/add_mesh_to_python.cpp:533]: We should do exactly the same I think.

Comment (roigcarlo) [reply kratos/python/add_mesh_to_python.cpp:533]: Done

**Review by rubenzorrilla — APPROVED** [review]

**Pull request merged**